packages/go_router/lib/src/delegate.dart

```diff
@@ -66,18 +106,37 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       pageKey: pageKey,
     );
 
-    _matches.push(newPageKeyMatch);
+    _matchList.push(newPageKeyMatch);
     notifyListeners();
   }
 
-  /// Returns `true` if there is more than 1 page on the stack.
+  /// Returns `true` if the active Navigator can pop.
   bool canPop() {
-    return _matches.canPop();
+    // Loop through navigators in reverse and call canPop()
+    final int matchCount = _matchList.matches.length;
+    for (int i = matchCount - 1; i >= 0; i -= 1) {
+      final RouteMatch match = _matchList.matches[i];
+      final RouteBase route = match.route;
+      if (route is GoRoute && route.parentNavigatorKey != null) {
+        final bool canPop = route.parentNavigatorKey!.currentState!.canPop();
+        // Similar to popRoute, it should not be possible for a GoRoute with
+        // parentNavigatorKey to be the only page, so canPop should return true
+        // in this case.
+        assert(canPop);
+        return canPop;
+      } else if (route is ShellRoute) {
+        final bool canPop = route.navigatorKey.currentState!.canPop();
+        if (canPop) {
+          return canPop;
+        }
```

Comment:
same thing for shellroute with parentNavigatorKey

Reply:
Same question as above

Reply:
adc610d1ccd749b6f70cbaad81983f5ad4160a2f